docs/docs/Tutorials/ROS2/CPPTemplate.md

```diff
@@ -0,0 +1,448 @@
+# Quick Start &mdash; Writing External Controller With GitHub Template Repository
+---
+
+In this tutorial, you will make and customize a GitHub repository from a GitHub Template with
+a ROS 2 C++ package and code ready to implement your own external controller utilizing the
+`buoy_api_cpp` interface. This interface may be used with the both the simulated and physical buoy.
+
+## Interfaces and Templates
+
+There are two GitHub
+[template repositories](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template)
+set up (C++/Python) for a quick start on writing a
+custom controller utilizing
+[buoy_api_cpp](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_cpp) and
+[buoy_api_py](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_py). Please see
+[C++ examples](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_cpp/examples) and
+[Python examples](https://github.com/osrf/buoy_msgs/tree/main/buoy_api_py/buoy_api/examples) for
+example controller implementations.
+
+* [mbari_wec_template_cpp](https://github.com/mbari-org/mbari_wec_template_cpp)
+* [mbari_wec_template_py](https://github.com/mbari-org/mbari_wec_template_py)
+
+---
+
+## Using C++ Template
+
+### Creating your own repo from the template
+
+You may also refer to GitHub's
+[template documentation](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template)
+
+To start using the C++ GitHub template
+
+1. Navigate to
+[mbari_wec_template_cpp](https://github.com/mbari-org/mbari_wec_template_cpp) and click the green
+button with the text `Use this template` and select `Create a new repository`
+
+    ![use_this_template](resources/use_this_template_cpp.png)
+
+2. Next, set up the repository like you would any new GitHub repository choosing the owner,
+repository name, public/private, etc.
+3. Make a ROS 2 workspace
+   ```
+   $ mkdir -p ~/buoy_ws/src
+   $ cd ~/buoy_ws/src
+   ```
+4. Now that your new repository is set up, clone it to your local machine, make a branch, etc.  
+   ```
+   $ git clone https://github.com/<owner>/<repo_name>.git
+   $ cd ~/buoy_ws
+   ```
+
+You should now have a C++ ROS 2 package with the following structure in your workspace `src`:
+
+```
+<repo_name>
+    ├── CMakeLists.txt
+    ├── config
+    │   └── controller.yaml
+    ├── include
+    │   └── mbari_wec_template_cpp
+    │       ├── controller.hpp
+    │       └── control_policy.hpp
+    ├── launch
+    │   └── controller.launch.py
+    ├── LICENSE
+    ├── package.xml
+    ├── README.md
+    └── src
+        └── controller.cpp
+```
+
```

Comment:
Suggestion on how to build the package could be useful too

Reply:
I have a section at the bottom for `Build, Test, Run`